### ISSUE TITLE
Sponge trouble in dumbbell parent domain

### ISSUE THREAD

**Issue by kshedstrom** [NOAA-GFDL/MOM6#1315]
I merged in the latest dev/gfdl for the sponges and got this:
```
MOM Day       0.000      0: En 0.000000E+00, MaxCFL  0.00000, Mass 1.117800000000E+17, Salt  35.00000000000, Temp  20.00000000000
    Total Energy: 0000000000000000  0.0000000000000000E+00
    Total Mass:   1.1178000000000000E+17, Change:   0.0000000000000000E+00 Error:  0.00000E+00 ( 0.0E+00)
    Total Salt:   3.9123000000000000E+15, Change:   0.0000000000000000E+00 Error:  0.00000E+00 ( 0.0E+00)
    Total Heat:   8.7747299999999999E+21, Change:   0.0000000000000000E+00 Error:  0.00000E+00 ( 0.0E+00)
At line 961 of file //import/c1/AKWATERS/kate/ESMG/ESMG-configs/src/MOM6/src/parameterizations/vertical/MOM_ALE_sponge.F90
Fortran runtime error: Index '32767' of dimension 1 of array 'cs' outside of expected range (2:1)

Error termination. Backtrace:
#0  0x11d2567 in __mom_ale_sponge_MOD_apply_ale_sponge
	at //import/c1/AKWATERS/kate/ESMG/ESMG-configs/src/MOM6/src/parameterizations/vertical/MOM_ALE_sponge.F90:961
#1  0x15d3227 in diabatic_ale_legacy
	at //import/c1/AKWATERS/kate/ESMG/ESMG-configs/src/MOM6/src/parameterizations/vertical/MOM_diabatic_driver.F90:996
#2  0x15d63df in __mom_diabatic_driver_MOD_diabatic
	at //import/c1/AKWATERS/kate/ESMG/ESMG-configs/src/MOM6/src/parameterizations/vertical/MOM_diabatic_driver.F90:377
#3  0x142a755 in step_mom_thermo
	at //import/c1/AKWATERS/kate/ESMG/ESMG-configs/src/MOM6/src/core/MOM.F90:1308
#4  0x14371a1 in __mom_MOD_step_mom
	at //import/c1/AKWATERS/kate/ESMG/ESMG-configs/src/MOM6/src/core/MOM.F90:821
#5  0xd42032 in mom_main
	at //import/c1/AKWATERS/kate/ESMG/ESMG-configs/src/MOM6/config_src/solo_driver/MOM_driver.F90:500
#6  0xd4318e in main
	at //import/c1/AKWATERS/kate/ESMG/ESMG-configs/src/MOM6/config_src/solo_driver/MOM_driver.F90:27
-------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code.. Per user-direction, the job has been aborted.
-------------------------------------------------------
```

**Comment by MJHarrison-GFDL** [comment]
Are we looking at this line: "tmp_val2(1:nz_data) = CS%Ref_val(m)%p(1:nz_data,c)"

https://github.com/NOAA-GFDL/MOM6/blob/dev/gfdl/src/parameterizations/vertical/MOM_ALE_sponge.F90#L961

**Comment by kshedstrom** [comment]
Yes, that line.

**Comment by kshedstrom** [comment]
CS%ref_val(1)%p is dimensioned (2,1200). 1:nz_data goes 1:32764.

**Comment by Hallberg-NOAA** [comment]
I think it very likely that this problem was caused when we merged in PR#1308 to dev/gfdl yesterday. Should we look at reverting that PR, or do you think a quick-fix is forthcoming?

Also, I am now finding in more extensive testing that PR#1308 changed the default value of SPONGE_IDAMP_VAR from "Idamp" to "IDAMP" without any discussion or heads-up. Could that be part of the issue here?

**Comment by MJHarrison-GFDL** [comment]
I have  [patch](https://github.com/NOAA-GFDL/MOM6/pull/1317) which I have confirmed works in the dumbbell parent case with no answer changes. @kshedstrom can confirm.  The change in the default "idamp" was upstream from the NCAR fork - @alperaltuntas can confirm if this is needed for any NCAR  test cases which may be using the default. I scanned the ESMg-config cases and it does not appear to be an issue, not is it an issue for MOM6-examples.
